### ISSUE TITLE
User defined objective function - question

### ISSUE THREAD

**Issue by rmldj** [dmlc/xgboost#15]
I have a question about the definition of the objective function. It is given in the demo.py example file as

``` python
# user define objective function, given prediction, return gradient and second order gradient
def logregobj( preds, dtrain ):
    labels = dtrain.get_label()
    grad = preds - labels
    hess = preds * (1.0-preds)
    return grad, hess
```

What is the objective function L(x_i,y_i) in this example as a function of current predictions x_i and the true labels y_i?

The `grad` is a vector which is the derivative dL/dx_i ? But I am confused about the meaning of Hessian here - here it is a vector while the Hessian is typically a matrix of second derivatives d^2L/dx_i dx_j. If it would be d^2L/dx_i^2 then I do not see what function L(x_i,y_i) would give

dL/dx_i =x_i-y_i
d^2L/dx_i^2= x_i (1-x_i)

as in the example above.
I would be grateful for a hint. I am not from the Machine Learning community so I may not know some obvious conventions. It would be helpful to update the wiki with 1-2 sentences more on this.

**Comment by Naireen** [comment]
so then the objective function is to left as "binary:logitraw", and then where do we pass in the custom loss function to the xgb model? I thought the custom loss function was to be set to the obj parameter?

**Comment by Jeannie0216** [comment]
I have a question about this code too. when I run the code on my computer, the python shows the error: Kernel died, restarting, do you know how can i fix it?

# user define objective function, given prediction, return gradient and second order gradient
def logregobj( preds, dtrain ):
    labels = dtrain.get_label()
    grad = preds - labels
    hess = preds * (1.0-preds)
    return grad, hess
